src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -48,6 +49,8 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
             services.AddSingleton<ITelemetryInitializer, WebSessionTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, WebUserTelemetryInitializer>();
 
+            services.AddInstance<IConfiguration>(config);
```

Comment:
Should I do something like this:
```
            services.AddSingleton<ITelemetryInitializer>(ComponentVersionTelemetryInitializer =>
            {
                return new ComponentVersionTelemetryInitializer(null, config);
            });
```
The problem I see is not having the httpContextAccessor